### ISSUE TITLE
[Epic] DAS Deprecation engineering tickets

### ISSUE THREAD

**Issue by aroltsch** [department-of-veterans-affairs/caseflow#9930]
Notes: 
1. Introduce new feature toggle `legacy_das_deprecation` and hide this code behind this feature. 
1. We will enable this feature to one judge group that doesn't have acting judges

1. At appeal automatic distribution, create JudgeAssignTask for a legacy appeal and set the appeal location to Caseflow in VACOLS. 
1. Judge's assign and review page must pull both types of cases: legacy appeals from Caseflow tasks table and legacy appeals from Vacols that match judge's location  
1. Create AttorneyTask when a judge assigns a legacy case to attorney only if the feature toggle is enabled and there exists a JudgeAssignTask for that appeal in Caseflow
1. Attorney's queue page must pull both types of cases: legacy appeals from Caseflow tasks table and legacy appeals from Vacols that match attorney's location
1. Create JudgeReviewTask when an attorney completes draft decision

**Comment by laurjpeterson** [comment]
**Caseflow location**
- Internal and external comms about adding another definition to the CASEFLOW location 

**Refactoring**
- Q: Is there any judge and attorney queue refactoring that would make sense to do throughout this, outside of replacing Legacy tasks with Caseflow tasks? If so, there might be more engineering tasks to incorporate.

**Attorney assign OMO task to judge**
- Perhaps the OMO task assigned to judge should be different than JudgeDecisionReviewTask, because at this point we know it's an OMO and not a draft decision. This allows a JudgeDecisionReviewTask to be created when the attorney actually writes a decision (and not an OMO request) for the case

**Judge assigns OMO tasks to VLJ Support**
- After judges sign OMO requests, a Caseflow task for OMO Request and put in VLJ Support Queue
   - What should this task be called on the front end? (Ask OMO support and VLJ support branch chief)
   - Can these tasks be part of round robin or not?
- OMO tasks must have same functionality as ColocatedTasks. They can be put on hold and completed. 
- When OMO tasks are marked complete, `JudgeAssignTask` is created and assigned back to the judge where it came from. Judge can then assign a task to an attorney to draft the decision. 

**Reporting**
- Re: "For VACOLS reporting purposes, create a DECASS record in Vacols with the following attributes:"
   - We should ask Jed and the pilot judge to help us test this
- OMOs have DAS records and attorneys get credit for them as a work product. Will we create a DECASS record in VACOLS fo when an OMO is sent to the judge or just for a draft decision? (I think we need both)

**Comment by aroltsch** [comment]
> Perhaps the OMO task assigned to judge should be different than JudgeDecisionReviewTask, because at this point we know it's an OMO and not a draft decision. This allows a JudgeDecisionReviewTask to be created when the attorney actually writes a decision (and not an OMO request) for the case

I've mentioned that here: "Create JudgeOmoAssignTask record when an attorney creates OMO and assigns it to a judge"

**Comment by aroltsch** [comment]
> When OMO tasks are marked complete, JudgeAssignTask is created and assigned back to the judge where it came from. Judge can then assign a task to an attorney to draft the decision.
At first I thought we could create a new task Type `JudgeOmoReviewTask` when a case comes back from OMO, but I actually like the idea of creating JudgeAssignTask

**Comment by aroltsch** [comment]
> After judges sign OMO requests, a Caseflow task for OMO Request and put in VLJ Support Queue

Would colocated users be working these cases?

**Comment by aroltsch** [comment]
> OMOs have DAS records and attorneys get credit for them as a work product. Will we create a DECASS record in VACOLS fo when an OMO is sent to the judge or just for a draft decision? (I think we need both)

Good point. We can create a DAS record when a judge creates an OMO request

**Comment by aroltsch** [comment]
> Is there any judge and attorney queue refactoring that would make sense to do throughout this, outside of replacing Legacy tasks with Caseflow tasks? If so, there might be more engineering tasks to incorporate.

Yes, these tickets imply refactoring.

**Comment by aroltsch** [comment]
If the case is randomly selected for a Quality Review, at what point should we create a decass record? 
After the QR is complete or when it is selected. What if the QR office sends the case back to the judge to correct things.

**Comment by aroltsch** [comment]
@laurjpeterson would you take another look at these comments and ticket description before tomorrow's spring planning please.

**Comment by laurjpeterson** [comment]
> After judges sign OMO requests, a Caseflow task for OMO Request and put in VLJ Support Queue
> Would colocated users be working these cases?

Some OMO folks will be joining the VLJ Support team. I think that those formerly-OMO users will continue to work these OMO tasks, but they are part of the VLJ Support team. 

I realize this could affect our round robin assignment.

**Comment by laurjpeterson** [comment]
> If the case is randomly selected for a Quality Review, at what point should we create a decass record? After the QR is complete or when it is selected. What if the QR office sends the case back to the judge to correct things.

I think it should be created when the case is sent to BVA dispatch to prevent any need to change the DAS record due to changes required by QR. So, when a BvaDispatch task is created, the DECASS record is created. Does this add complexity, though?
